packages/dbgpt-app/src/dbgpt_app/openapi/api_v1/agentic_data_api.py

```
@@ -2212,23 +2282,45 @@ def advance_todo_list() -> Optional[List[Dict[str, str]]]:
-                Terminate(),
-            ]
-            + business_tools
-            + connector_tool_extras
-        )
+        if tool_mode == "knowledge":
+            # Knowledge-chat mode: only kb tools + todowrite + terminate.
+            # No skill/shell/sql/html/code tools to keep the agent focused.
+            tool_pack = ToolPack(
+                [todowrite_tool, question_tool, Terminate()]
+                + business_tools
+                + connector_tool_extras
```

Ruff (0.15.21)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)